tests/monitor/test_data_monitoring.py

```diff
@@ -68,6 +77,24 @@ def execute_query_side_effect(*args, **kwargs):
     return snowflake_data_mon
 
 
+@pytest.fixture
+def snowflake_data_monitoring_slack_workflow(slack_workflows_config_mock, snowflake_con_mock, dbt_runner_mock):
+    # This cursor mock will use the side effect to return non empty configuration
+    snowflake_cursor_context_manager_return_value = snowflake_con_mock.cursor.return_value.__enter__.return_value
```

Comment:
Instead of duplicating the 'snowflake_data_monitoring' fixture's code, you could create a regular function that is called 'create_data_monitoring_mock' that can get the config as a parameter. Then both the fixture 'snowflake_data_monitoring' and the fixture 'snowflake_data_monitoring_slack_workflow' can use it to create a snowflake_data_mon

Reply:
[consolidate mock generation code to one function taking slack config as param](https://github.com/elementary-data/elementary/pull/54/commits/b91ceb722fc3f0f1919c12893df9734551e54164)